src/main/java/jenkins/scm/api/SCMFile.java

```diff
@@ -78,6 +78,7 @@ protected SCMFile() {
      * Constructor for any entry that is not the root.
      *
      * @param parent the parent reference or {@code null} if this is the root object.
+     * @param name   the name of this entry.
```

Comment:
So is `/` illegal here? If so, a `child` implementation will be obliged to construct intermediate `SCMFile` directory objects, which seems onerous.

IMO rather than a `parent` field there should just be a backreference to the `SCMFileSystem` root, and a full `String path`. `getParent` can then simply construct a new flyweight object by stripping off the last path component.